api/maven-api-core/src/main/java/org/apache/maven/api/Constants.java

```diff
@@ -0,0 +1,239 @@
+     * </ul>
+     * Example filter expression: <code>"h(5);s;e(org.foo:bar:1)</code> will cause: ranges are filtered for "top 5" (instead
+     * full range), snapshots are banned if root project is not a snapshot, and if range for <code>org.foo:bar</code> is
+     * being processed, version 1 is omitted.
+     * </p>
+     *
+     * @since 4.0.0
+     */
+    @Config
+    public static final String MAVEN_VERSION_FILTERS = "maven.versionFilters";
+
+    /**
+     * User property for chained LRM: list of "tail" local repository paths (separated by comma), to be used with
+     * {@code org.eclipse.aether.util.repository.ChainedLocalRepositoryManager}.
+     * Default value: <code>null</code>, no chained LRM is used.
+     *
+     * @since 3.9.0
+     */
+    @Config
+    public static final String MAVEN_REPO_LOCAL_TAIL = "maven.repo.local.tail";
+
+    /**
+     * User property for reverse dependency tree. If enabled, Maven will record ".tracking" directory into local
+     * repository with "reverse dependency tree", essentially explaining WHY given artifact is present in local
+     * repository.
+     * Default: <code>false</code>, will not record anything.
+     *
+     * @since 3.9.0
+     */
+    @Config(defaultValue = "false")
+    public static final String MAVEN_REPO_LOCAL_RECORD_REVERSE_TREE = "maven.repo.local.recordReverseTree";
+
+    /**
+     * User property for selecting dependency manager behaviour regarding transitive dependencies and dependency
+     * management entries in their POMs. Maven 3 targeted full backward compatibility with Maven2, hence it ignored
+     * dependency management entries in transitive dependency POMs. Maven 4 enables "transitivity" by default, hence
+     * unlike Maven2, obeys dependency management entries deep in dependency graph as well.
+     * <p>
+     * Default: <code>"true"</code>.
+     * </p>
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "true")
+    public static final String MAVEN_RESOLVER_DEPENDENCY_MANAGER_TRANSITIVITY =
+            "maven.resolver.dependencyManagerTransitivity";
+
+    /**
+     * Resolver transport to use.
+     * Can be <code>default</code>, <code>wagon</code>, <code>apache</code>, <code>jdk</code> or <code>auto</code>.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "default")
+    public static final String MAVEN_RESOLVER_TRANSPORT = "maven.resolver.transport";
+
+    /**
+     * Plugin validation level.
+     *
+     * @since 3.9.2
+     */
+    @Config(defaultValue = "inline")
+    public static final String MAVEN_PLUGIN_VALIDATION = "maven.plugin.validation";
+
+    /**
+     * Plugin validation exclusions.
+     *
+     * @since 3.9.6
+     */
+    @Config
+    public static final String MAVEN_PLUGIN_VALIDATION_EXCLUDES = "maven.plugin.validation.excludes";
+
+    /**
+     * ProjectBuilder parallelism.
+     *
+     * @since 4.0.0
+     */
+    @Config(type = "java.lang.Integer", defaultValue = "cores/2 + 1")
```

Comment:
How does this `expr` work?

Reply:
`defaultValue` is documentation only

Reply:
Ah OK, I was really suprised that we have this magic now...